core/src/test/java/com/linecorp/armeria/client/TestReconfigurableServer.java

```diff
@@ -0,0 +1,206 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+class TestReconfigurableServer {
+
+    @Test
+    void shouldBeAbleToReconfigureServer() throws Exception {
+        final ServerBuilder sb = Server.builder();
+        sb.http(9009);
+
+        sb.service("/test1", (ctx, req) -> HttpResponse.of("Hello, world!"));
+
+        sb.service("/test11/{name}", (ctx, req) -> {
+            final String param = ctx.pathParam("name");
+            return HttpResponse.of("Hello, " + param.toUpperCase());
+        });
+
+        final Server server = sb.build();
+
+        server.start().join();
+
+        final WebClient originalClient = WebClient.of("http://localhost:" + server.activeLocalPort());
+        final AggregatedHttpResponse response = originalClient.get("/test1").aggregate().join();
+        assertThat(response.status()).isEqualTo(HttpStatus.OK);
+        assertThat(response.contentUtf8()).isEqualTo("Hello, world!");
+
+        final AggregatedHttpResponse response11 = originalClient.get("/test11/world").aggregate().join();
+        assertThat(response11.status()).isEqualTo(HttpStatus.OK);
+        assertThat(response11.contentUtf8()).isEqualTo("Hello, WORLD");
+
+        Thread.sleep(1000); // Some time goes by now want to reconfigure new server
+        System.out.println("Configuring new server");
+        server.reconfigure(serverBuilder -> {
+            // Replace the entire routes with the following two services.
+            serverBuilder.service("/test2", (ctx, req) -> HttpResponse.of("Hello, world!"));
+
+            serverBuilder.service("/test2/{name}",
+                    (ctx, req) ->
+                            HttpResponse.of("Hello, " + ctx.pathParam("name").toUpperCase()));
+        });
+
+        // retry calling the original service. Should return 404.
+        final AggregatedHttpResponse response1 = originalClient.get("/test11/world").aggregate().join();
+        assertThat(response1.status()).isEqualTo(HttpStatus.NOT_FOUND);
+
+        final AggregatedHttpResponse res = originalClient.get("/test2").aggregate().join();
+        assertThat(res.status()).isEqualTo(HttpStatus.OK);
+        assertThat(res.contentUtf8()).isEqualTo("Hello, world!");
+
+        //Open new connection to check if the newly reconfigured server is now operational
```

Comment:
```suggestion
        // Open new connection to check if the newly reconfigured server is now operational
```

Reply:
Done

Reply:
I ran checkstyl on my local and it passed. How do I catch these problems before I submit the CR?